### PR TITLE
Engagement banner copy test 2

### PR DESCRIPTION
**Pull request by rupertbates**
## What does this change?
Extends the membership engagement banner copy test, adds new message variants and uses the new prominent banner style

Tracking codes are defined [here](https://docs.google.com/spreadsheets/d/1ymFXK9eSf2TV_jjniaCdrS2omzAWjTHUHKobSj4Ceqo/edit?ts=58246161#gid=0)
New text is [here](https://docs.google.com/document/d/16kbjk6yp96SS9sYa7z88yPdmjlZHuSfnjE2JxOQz834/edit?ts=58245f54)

## What is the value of this and can you measure success?
We hope to discover the most effective copy to use to increase membership sign-ups.

## Does this affect other platforms - Amp, Apps, etc?

No web only

## Request for comment
@rtyley @Ap0c

### CODE REVIEW

File: static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner.js [hunk 1]
```diff
@@ -63,34 +61,25 @@ define([
     };
 
     function formatEndpointUrl(edition, message) {
-        return endpoints[edition] + '?INTCMP=' + message.campaign;
+        return endpoints[edition] + '?INTCMP=' + message.campaign + "_prominent";
```
Comment (jfsoul) [inline static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner.js]: looks like linting is failing due to use of double quotes here

File: static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner.js [hunk 2]
```diff
@@ -110,61 +99,59 @@ define([
...
             thisColour = thisInstanceColour(colours);
-            cssModifierClass = 'membership-' + prominentMarker + ' ' + thisColour;
+            cssModifierClass = 'membership-prominent ' + thisColour;
             customOpts.execute = function () {
                 var buttonCaption = $('#membership__engagement-message-button-caption'),
                     buttonEl = $('#membership__engagement-message-button');
                 fastdom.write(function () {
                     buttonEl.removeClass('is-hidden');
-                    buttonEl.addClass(prominentMarker);
+                    buttonEl.addClass("prominent");
```
Comment (jfsoul) [inline static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner.js]: double quotes again

File: static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner.js [hunk 3]
```diff
@@ -63,34 +61,25 @@ define([
...
-            }
-
-            if (opts.testName === 'messaging-test-us-1') {
-                var usEngagementText = $('.site-message__message.site-message__message--membership');
-                if (usEngagementText && opts.setEngagementText) {
-                    usEngagementText[0].textContent = opts.setEngagementText;
-                }
+        if (opts.setEngagementText) {
+            var engagementText = $('#membership__engagement-message-text');
+            if (engagementText && opts.setEngagementText) {
```
Comment (jfsoul) [inline static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner.js]: already checked `opts.setEngagementText` is truthy above

File: static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner.js [hunk 4]
```diff
@@ -110,61 +99,59 @@ define([
...
-                    setEngagementText: messagingTestVariant === 'control' ? undefined : variantMessages[messagingTestVariant]
-                };
-                customJs = getCustomJs;
-            }
+        if (messagingTestVariant && messagingTestVariant !== 'notintest') {
+            var variantMessages = {
+                Get_round_to: 'Not got round to supporting us yet? Now is the time. Give £5 a month today',
+                Give_upfront: 'Give £5 a month to help support our journalism and make the Guardian\'s future more secure',
+                Together_informed: 'Together we can keep the world informed. Give £5 a month to support our journalism',
+                Coffee_5: 'For less than the price of a coffee a week, you could help secure the Guardian’s future. Support our journalism for £5 a month'
```
Comment (jfsoul) [inline static/src/javascripts/projects/common/modules/commercial/membership-engagement-banner.js]: the apostrophe used here is different to the single quote used in `Give_upfront`

**Review by jfsoul — APPROVED** [review]
Minor comments aside, looks good to me (might want to get membership opinion of course). Let's get this merged so the old switch doesn't break master 👍

**Comment by philwills** [comment]
@rupertbates @rtyley @Ap0c 

Is this ready to be merged? The old test has now expired and is blocking builds.

**Comment by jfsoul** [comment]
Thanks @rupertbates 👏

**Pull request merged**